pkgs/development/haskell-modules/with-packages-wrapper.nix

```diff
@@ -102,6 +107,9 @@ buildEnv {
     done
 
     ${lib.optionalString hasLibraries "$out/bin/${ghcCommand}-pkg recache"}
+    ${lib.optionalString (isGhcjs && ghcLibdir != null) ''
+      mv '${libDir}/ghc_libdir_override' '${libDir}/ghc_libdir'
```

Comment:
While I'm still not sure what we're trying to accomplish here: is there any reason not to inline the `echo '${ghcLibdir}' > '${libDir}/ghc_libdir_override'` here, instead of putting `ghc_libdir_override` in the symlink forest and then `mv`ing it over?